Tests/Marketplace/upload_packs.py

```diff
@@ -78,25 +78,26 @@ def download_and_extract_index(storage_bucket, extract_destination_path):
     Returns:
         str: extracted index folder full path.
         Blob: google cloud storage object that represents index.zip blob.
+        str: downloaded index generation.
 
     """
     index_storage_path = os.path.join(GCPConfig.STORAGE_BASE_PATH, f"{GCPConfig.INDEX_NAME}.zip")
     download_index_path = os.path.join(extract_destination_path, f"{GCPConfig.INDEX_NAME}.zip")
 
     index_blob = storage_bucket.blob(index_storage_path)
     index_folder_path = os.path.join(extract_destination_path, GCPConfig.INDEX_NAME)
+    index_generation = 0  # Setting to 0 makes the operation succeed only if there are no live versions of the blob
 
     if not os.path.exists(extract_destination_path):
         os.mkdir(extract_destination_path)
 
     if not index_blob.exists():
         os.mkdir(index_folder_path)
-        return index_folder_path, index_blob
+        return index_folder_path, index_blob, index_generation
 
-    # index zip should never be cached in the memory, should be updated version
-    index_blob.cache_control = "no-cache,max-age=0"
     index_blob.download_to_filename(download_index_path)
```

Comment:
I think this should be done after `reload` and use `if_generation_match` to make sure the version we download is the generation number we saved.

Reply:
@glicht Switched line after `reload` but I don't see that `download_to_filename` function has `if_generation_match` in `google-cloud-storage` library version that we use (although in official docs there is such an argument).
<img width="711" alt="Screen Shot 2020-07-19 at 0 00 56" src="https://user-images.githubusercontent.com/45535078/87861865-23e68580-c953-11ea-97a4-fd0a1dbf87c0.png">

Reply:
I think you are looking at old code. v1.29.0 has it. See here: https://github.com/googleapis/python-storage/blob/v1.29.0/google/cloud/storage/blob.py#L956